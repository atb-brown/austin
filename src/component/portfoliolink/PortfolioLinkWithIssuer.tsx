import Link from "../Link";
import { PortfolioLinkProps } from "./PortfolioLinkProps";
import React from "react";

/**
 * A link to some sort of portolio item that is not issued by an organization.
 *
 * @param {PortfolioLinkProps} props Properties.
 * @return {ReactElement} The portfolio link component.
 */
export default function LinkWithoutIssuer({
  description,
  urlId,
  url,
}: PortfolioLinkProps) {
  return (
    <div>
      <div>{description}</div>
      <div>
        <Link description={urlId} url={url} />
      </div>
      <br />
    </div>
  );
}
