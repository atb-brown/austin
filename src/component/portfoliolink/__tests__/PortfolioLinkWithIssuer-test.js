import PortfolioLinkWithIssuer from "../PortfolioLinkWithIssuer";
import renderer from "react-test-renderer";

it("renders to match snapshot", () => {
  const tree = renderer
    .create(
      <PortfolioLinkWithIssuer
        description="description Field"
        url="https://example.com"
        urlId="linkId Field"
        issuerDescription="issuer description field"
        issuerUrl="https://https://github.com/atb-brown"
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
