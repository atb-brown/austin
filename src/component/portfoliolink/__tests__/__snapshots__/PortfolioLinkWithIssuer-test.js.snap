// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<div
  className="App-text"
>
  <div>
    description Field
  </div>
  <div>
    <a
      className="App-link"
      href="https://example.com"
    >
      linkId Field
    </a>
  </div>
  <br />
</div>
`;
